csrc/quantization/gptq_marlin/gptq_marlin.cu

```diff
@@ -130,7 +134,13 @@ __device__ inline uint32_t prmt(uint32_t a) {
 // values. We mostly follow the strategy in the link below, with some small
 // changes:
 // https://github.com/NVIDIA/FasterTransformer/blob/main/src/fastertransformer/cutlass_extensions/include/cutlass_extensions/interleaved_numeric_conversion.h
-__device__ inline FragB dequant_4bit(int q) {
+template <typename scalar_t>
+__device__ inline typename ScalarType<scalar_t>::FragB dequant_4bit(int q) {
+  throw std::runtime_error("unsupported");
```

Comment:
I'm not sure what the standard is but I think most checks in the code use `TORCH_CHECK` rather than `throw`.